src/averageAges.js

```diff
@@ -37,7 +45,20 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const onlyWomen = people.filter(({ name, sex }) => {
+    const hasChildren = !withChildren
+      || people.some(person => person.mother === name);
+
+    return sex === 'f' && hasChildren;
+  });
+
+  const sumAgesWomen = onlyWomen.reduce((sum, woman) => {
+    const womanAge = woman.died - woman.born;
+
+    return sum + womanAge;
+  }, 0);
```

Comment:
the functions for calculating ages sum for men and women are the same, let's create universal function that will accept the array of people and will return the sum of their ages

Reply:
Roger that, thanks for the hint =)